Sources/XcodeGraphMapper/Mappers/Targets/PBXTargetMapper.swift

```diff
@@ -331,6 +353,179 @@ struct PBXTargetMapper: PBXTargetMapping {
+            )
+            .map {
+                ResourceFileElement(path: $0)
+            }
+            resources.append(contentsOf: groupResources)
+        }
+
+        return resources
+    }
+
+    private func fileSystemSynchronizedGroupsFrameworks(
+        from pbxTarget: PBXTarget,
+        xcodeProj: XcodeProj
+    ) async throws -> [TargetDependency] {
+        let fileSystemSynchronizedGroups = pbxTarget.fileSystemSynchronizedGroups ?? []
+        var frameworks: [TargetDependency] = []
+        for fileSystemSynchronizedGroup in fileSystemSynchronizedGroups {
+            guard let path = fileSystemSynchronizedGroup.path else { continue }
+            let directory = xcodeProj.srcPath.appending(component: path)
+            let membershipExceptions = Set(
+                fileSystemSynchronizedGroup.exceptions
+                    .map { $0.compactMap(\.membershipExceptions).flatMap { $0 } } ?? []
+            )
+
+            let attributesByRelativePath = fileSystemSynchronizedGroup.exceptions?.reduce([:]) { acc, element in
+                acc.merging(element.attributesByRelativePath ?? [:], uniquingKeysWith: { $1 })
+            }
+
+            let groupFrameworks: [TargetDependency] = try await globFiles(
+                directory: directory,
+                include: [
+                    "**/*.framework",
+                ],
+                membershipExceptions: membershipExceptions
+            )
+            .map {
+                return .framework(
+                    path: $0,
+                    status: attributesByRelativePath?[$0.relative(to: directory).pathString]?
+                        .contains("Weak") == true ? .optional : .required,
+                    condition: nil
+                )
+            }
+            frameworks.append(contentsOf: groupFrameworks)
+        }
+
+        return frameworks
+    }
+
+    private func addHeadersFromFileSystemSynchronizedGroups(
+        from pbxTarget: PBXTarget,
+        xcodeProj: XcodeProj,
+        headers: Headers?
+    ) async throws -> Headers? {
+        let fileSystemSynchronizedGroups = pbxTarget.fileSystemSynchronizedGroups ?? []
+        var publicHeaders = headers?.public ?? []
+        var privateHeaders = headers?.private ?? []
+        var projectHeaders = headers?.project ?? []
+        for fileSystemSynchronizedGroup in fileSystemSynchronizedGroups {
+            guard let path = fileSystemSynchronizedGroup.path else { continue }
+            let directory = xcodeProj.srcPath.appending(component: path)
+            for synchronizedBuildFileSystemExceptionSet in fileSystemSynchronizedGroup.exceptions ?? [] {
+                for publicHeader in synchronizedBuildFileSystemExceptionSet.publicHeaders ?? [] {
+                    publicHeaders.append(directory.appending(component: publicHeader))
+                }
+                for privateHeader in synchronizedBuildFileSystemExceptionSet.privateHeaders ?? [] {
```

Comment:
Can't exceptions be defined for `projectHeaders?

Reply:
Initially, I couldn't figure out a way how to do it, but it is actually possible once you add a headers build phase. I added this logic in 👍